src/http-gateways/trustless-gateway.md

```diff
@@ -302,31 +302,73 @@ of their presence in the DAG or the value assigned to the "dups" parameter, as
 
-The `meta=eof` parameter allows clients to request the server to include additional metadata about the
-CAR to be included at the end of the response body.
+The `meta` parameter allows clients to request the server to include additional metadata about the CAR along with the response body.
 
-This parameter SHOULD only be used with CAR `version=1`.
-Values other than `eof` SHOULD be ignored.
+The value of this parameter includes both the location where the metadata is given (e.g. `eof`) as well as the type of data received (e.g. `json`) separated by a `+`, to give a value such as `meta=eof+json`
 
-When the parameter is not set, the server must not add any extra CAR blocks to the response.
+When the location parameter is set to `eof`, which is currently the only supported value, the server SHOULD respond with <Response body as CARv1 stream> <0x00 byte> <Metadata>.
 
-The metadata block is a regular CAR block with the following properties:
+The only supported value for the data type parameter is `json`. This signifies that the metadata MUST be a JSON object.
 
-- CID specifies multicodec `car-metadata` (`0x04ff`), see
-  [multicodec#334](https://github.com/multiformats/multicodec/pull/334).
+This parameter MUST only be used with CAR `version=1`.
 
-- The payload contains metadata encoded as DAG-CBOR.
+When the parameter is not set or does not equal `eof+json`, the server SHOULD not add any extra blocks to the response, neither the 0x00 byte nor any metadata.
 
-The metadata MUST include the following fields:
+When `meta=eof+json`, the JSON object SHOULD conform to the following [JSON schema](https://json-schema.org/).
 
-- `len` - byte length of the CAR data (excluding the metadata block)
-- `b3h` - Blake3 hash (checksum) of the CAR data (excluding the metadata block).
-- `b3h_sig` - A signature over `<len><b3h><request>` using server's Ed2559 identity.
-  - `len` is encoded as `varint`,
-  - `b3h` is encoded as 32 bytes,
-  - The effective query as executed by the gateway. This query is the request url - path and query string arguments.
+```json
+{
+  "properties": {
+    "description": "Properties of the response"
+    "type": "object"
+  },
+  "error": {
+    "description": "Error message"
+    "type": "string"
+  },
+  "sig": {
+    "description": "A signature, using the server's Ed2559 identity, over the metadata properties object"
+    "type": "string"
+  }
+}
+```
+
+The properties object can include any fields that the server would like to implement. The following properties fields are mentioned explicitly to reach a convention on their definition as they have existing use cases.
+
+```json
+{
+  "car_bytes": {
+    "description": "The total byte length of the CAR stream (excluding the 0x00 byte and the metadata block)",
+    "type": "integer"
+  },
+  "data_bytes": {
+    "description": "Total byte length of blocks (excluding the 0x00 byte and the metadata block, but including duplicates when present)",
```

Comment:
I think the intention behind `data_bytes` is to describe the length of the original content before it was encoded into a CAR file. I.e. if you take a cat picture and upload it to IPFS, then `car_bytes` is the length of the CAR file created from that picture and `data_bytes` is the length of the picture file.

Reply:
ah ok, I will fix this